### PR TITLE
Move the most of metadata into `PEP 621`-compliant `pyproject.toml`.

### PR DESCRIPTION
**Pull request by KOLANICH**
Requires #1005

### CODE REVIEW

**Comment by vnmabus** [comment]
It partially overlaps with #975, I think.

**Comment by MartinBubel** [comment]
I like it! What do you think @ekalosak @lawrennd ? Should merge this (after resolving conflicts etc) into the upcoming release or keep it for the subsequent one with more updates/qof-fixes?

**Comment by lawrennd** [comment]
I'd be supportive of this.

**Comment by ekalosak** [comment]
Yes pls update the branch and we need to merge it.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+[build-system]
+requires = ["setuptools>=61.2", "numpy>=1.7",]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "GPy"
+authors = [{name = "GPy Authors: https://github.com/SheffieldML/GPy/graphs/contributors", email = "gpy.authors@gmail.com"},]
+license = {text = "BSD-3-Clause"}
+description = "The Gaussian Process Toolbox"
+readme = "README.rst"
```
Comment (MartinBubel) [inline pyproject.toml:10]:
```suggestion
readme = "README.md"
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
```
Comment (MartinBubel) [inline pyproject.toml:22]:
```suggestion
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
    "Programming Language :: Python :: 3.13",
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    "Topic :: Software Development",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "numpy>=1.7",
+    "six",
+    "paramz>=0.9.0",
+    "cython>=0.29",
+    "scipy>=1.3.0; python_version >= '3.6'",
+    "scipy>=1.3.0,<1.5.0; python_version < '3.6'",
```
Comment (MartinBubel) [inline pyproject.toml:35]:
```suggestion
    "scipy>=1.3.0,<=1.12.0"
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    "Framework :: IPython",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "numpy>=1.7",
+    "six",
+    "paramz>=0.9.0",
```
Comment (MartinBubel) [inline pyproject.toml:32]:
```suggestion
    "paramz>=0.9.6",
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Download = "https://github.com/SheffieldML/GPy/archive/refs/heads/devel.zip"
+"Source Code" = "https://github.com/SheffieldML/GPy"
+"Bug Tracker" = "https://github.com/SheffieldML/GPy/issues"
+
+[project.optional-dependencies]
+docs = ["sphinx"]
+optional = ["mpi4py", "ipython>=4.0.0"]
+plotting = [
+    "matplotlib==3.3.4; python_version >= '3.6'",
+    "matplotlib==3.0.0; python_version < '3.6'",
```
Comment (MartinBubel) [inline pyproject.toml:50]:
```suggestion
    "matplotlib>=3.3.4",
```

**Review by MartinBubel — CHANGES_REQUESTED** [review]
LGTM, only some minor things to be done

**Pull request merged**